src/paho/mqtt/client.py

```diff
@@ -1058,7 +1089,7 @@ def username_pw_set(self, username, password=None):
         username: The username to authenticate with. Need have no relationship to the client id.
         password: The password to authenticate with. Optional, set to None if not required.
         """
-        self._username = username.encode('utf-8')
```

Comment:
This is as it was in `fixes`.
But I agree with you: the handling of the unicode encoding/decoding is far from consistent.